src/main/java/org/sunbird/progress/service/MandatoryContentServiceImpl.java

```diff
@@ -150,6 +151,8 @@ public void enrichProgressDetails(String authUserToken, MandatoryContentResponse
 
 	public Map<String, Object> getUserProgress(SunbirdApiRequest requestBody, String authUserToken, String rootOrgId, String userChannel) {
```

Comment:
the existing API seems like returning all batches from course if possible... 
if course has 3 batches and each batch has 5 users -- this API would return 15 users details... 

the new design tries to get users from specific course batch -- and tries to provide pagination... 
do add a v2/progress/getUserProgressByBatch method and copy this updated implementation to new method... 

do get rid of the for loop for the batch in new method. make sure validation done for all newly added fields from requestBody....